### DISCUSSION TITLE
Trying to detect snmp but no response

### DISCUSSION THREAD

**Discussion by No-Github** [projectdiscovery/nuclei#1474] [Q&A] [answered]
### Nuclei version:

```
2.5.3
```

### Current Behavior:

```
debug No output
```

### Expected Behavior:

```
Output request, response
```

### Steps To Reproduce:

```
┌──(root💀kali)-[~]
└─# cat snmp-detect.yaml            
id: snmp-detect

info:
  name: snmp-detect
  author: ffffffff0x
  severity: info
  tags: network,service,snmp

network:
  - host:
      - "{{Hostname}}"
      - "{{Hostname}}:161"

    matchers:
      - type: word
        words:
          - "snmp"                                                                                                                                                                                
```
```
┌──(root💀kali)-[~]
└─# nuclei -t snmp-detect.yaml -duc -debug -u xx.xx.xx.xx:161

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.5.3

		projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] Using Nuclei Engine 2.5.3
[INF] Using Nuclei Templates 
[INF] Using Interactsh Server https://interactsh.com
[INF] Templates added in last update: 0
[INF] Templates loaded for scan: 1
[INF] No results found. Better luck next time!
```

![image](https://user-images.githubusercontent.com/18167071/144372801-f4a944ad-07b1-41e0-b2a6-1a1f391fa7bc.png)
![image](https://user-images.githubusercontent.com/18167071/144372880-11bc4100-d60c-4263-b941-f6b5a1708832.png)

**Comment by ehsandeep** [comment]
@No-Github you need to send input data in order to get a response from this service.

**Comment by No-Github** [comment]
I changed the template, but it still gives the same result

```
id: snmp-detect

info:
  name: snmp-detect
  author: ffffffff0x
  severity: info
  tags: network,service,snmp

network:
  - inputs:
      - data: "005056fbd80d000c292406ec080045000047e6674000401189f4c0a88d81d39da882e2f600a1003383db302902010004067075626c6963a01c020401351ccc020100020100300e300c06082b060102010105000500"

  - host:
      - "{{Hostname}}"
      - "{{Host}}:161"

    matchers:
      - type: word
        words:
          - "snmp"
```

![image](https://user-images.githubusercontent.com/18167071/149256945-a08e438d-da1b-49cf-865a-1aa0ba063349.png)


```
id: snmp-detect

info:
  name: snmp-detect
  author: ffffffff0x
  severity: info
  tags: network,service,snmp

network:
  - inputs:
      - data: "\r\n"

  - host:
      - "{{Hostname}}"
      - "{{Host}}:161"

    matchers:
      - type: word
        words:
          - "snmp"
```

![image](https://user-images.githubusercontent.com/18167071/149257045-7975ecaa-a661-4568-837c-cc844c07f9eb.png)

![image](https://user-images.githubusercontent.com/18167071/149257105-372b7f73-8185-4074-80cc-84a3da795f85.png)

![image](https://user-images.githubusercontent.com/18167071/149257195-71a27bf6-689c-4da1-8a61-6d4f024aaa95.png)

I guess it's because the target SNMP only responds to UDP, but nuclei sends tcp requests

Do you have a solution?

**Comment by forgedhallpass** [comment, marked as answer]
@No-Github as you have correctly pointed out already, SNMP by default uses UDP, which is not currently supported by nuclei.
You can track the feature request under https://github.com/projectdiscovery/nuclei/issues/2129

**Comment by geeknik** [comment]
I thought nuclei had support for UDP already in place? AFAIK, the `DNS` templates use UDP. If I run `nuclei -t integration_tests/dns/basic.yaml -u google.com` while running tcpdump like `sudo su -c "tcpdump -vv -i eth0 udp 'port 53'"`, I can see the UDP requests:

```
08:10:34.109534 IP (tos 0x0, ttl 64, id 36528, offset 0, flags [DF], proto UDP (17), length 67)
    box.local.48701 > one.one.one.one.domain: [bad udp cksum 0xe11d -> 0x4d06!] 54895+ [1au] A? google.com. ar: . OPT UDPsize=4096 (39)
08:10:34.171540 IP (tos 0x0, ttl 58, id 14046, offset 0, flags [DF], proto UDP (17), length 83)
    one.one.one.one.domain > box.local.48701: [udp sum ok] 54895 q: A? google.com. 1/0/1 google.com. A 142.250.80.14 ar: . OPT UDPsize=1232 (55)
```

**Reply by forgedhallpass** [reply]
That may be true, but the `network` attribute doesn't support UDP yet.
